conversion.go

```diff
@@ -0,0 +1,129 @@
+package safecast
+
+import (
+	"errors"
+	"fmt"
+	"math"
+)
+
+type Integer interface {
+	~int | ~uint | ~int8 | ~uint8 | ~int16 | ~uint16 | ~int32 | ~uint32 | ~int64 | ~uint64
+}
+
+var ErrOutOfRange = errors.New("out of range")
```

Comment:
might be more complex but could be useful for callers to say more, ie value and range or type

ie make an error type with a few more fields and a .Error()?

Reply:
I opened https://github.com/ccoVeille/go-safecast/issues/15